[Samuel Mui] iP

samuelmui8 opened the pull request:
# Brobot
> "Because brothers don't let each other wander in the dark alone" - Jolene Perry

Trying to remember tasks alone? Fret not. Brobot is like your bro but he helps you to remember things. He's
- text-based
-  ~easy~ **super easy** to use
- **a bro**

All you need to do is,
1. Download the jar file from [here](https://github.com/samuelmui8/ip/releases/download/v0.1-Duke.jar/duke-0.1.jar).
2. Copy the jar file into an empty folder.
3. Open a command window in that folder.
4. Run the command java -jar {filename}.jar e.g., java -jar Duke.jar (i.e., run the command in the same folder as the jar file).
5. Add your tasks.
6. Let Brobot manage your tasks for you! 😺 

Features:
- [x] Managing todos
- [x] Managing deadlines
- [x] Managing events
- [x] Search for tasks
- [ ] Graphical User Interface (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
    Application.launch(Main.class, args);
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

samuelmui8 committed:
changed Duke to Brobot

samuelmui8 committed:
added echo and exit behaviour

samuelmui8 committed:
added list

samuelmui8 committed:
added ability to mark and unmark tasks

samuelmui8 committed:
added todos, events, deadlines

samuelmui8 committed:
added automated text ui testing

samuelmui8 committed:
added error handling

samuelmui8 committed:
added support for deleting tasks

samuelmui8 committed:
added enums and used switch

samuelmui8 committed:
add method for each bot command

samuelmui8 committed:
add support to save list

samuelmui8 committed:
Merge branch 'branch-level-7'

samuelmui8 committed:
Change date fields in Tasks from String to LocalDate

samuelmui8 committed:
Merge branch 'branch-level-8'

samuelmui8 committed:
Extract closely related code as classes

samuelmui8 committed:
Merge branch 'branch-A-MoreOOP'

samuelmui8 committed:
divide classes into packages

samuelmui8 committed:
Merge branch 'branch-A-Packages'

samuelmui8 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

samuelmui8 committed:
Add junit tests for Deadline, Parser and TaskList

samuelmui8 committed:
Merge branch 'A-JUnit'

samuelmui8 committed:
Add javadocs for remaining methods

samuelmui8 committed:
Add finding tasks by searching for keywords

samuelmui8 committed:
Refactor code to adhere to coding standard

samuelmui8 committed:
Merge branch 'branch-A-JavaDoc'

samuelmui8 committed:
Merge branch 'branch-A-CodingStandard'

samuelmui8 committed:
Merge branch 'branch-Level-9'

samuelmui8 committed:
Add checkstyle, fix style violations and minor bugs

samuelmui8 committed:
Merge branch 'branch-A-CheckStyle'

samuelmui8 committed:
Add JavaFX as a GUI

samuelmui8 committed:
Merge branch 'branch-Level-10'

mamayuan commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,17 @@
+    MARK,
+    UNMARK,
+    DELETE,
+    TODO,
```
Since the enum class name is Command, maybe you can consider naming the elements with verbs? now it looks like it is mixed with nouns, verbs and also adjectives

mamayuan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,115 @@
+    public static Task stringToTask(String string) throws DukeException {
+        String[] words = string.split("\\s\\|\\s");
+        String taskType = words[0];
+        boolean done;
```
Based on what we have learnt, maybe change it to isDone for a boolean value?

mamayuan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,80 @@
+        }
+
+    }
+
```
some unnecessary whitespace here~

mamayuan commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,245 @@
+        try {
+            int index = Integer.parseInt(taskIndex) - 1;
+            list.get(index);
+            StringBuilder s = new StringBuilder("Noted! I've deleted this task from the list:\n"
```
perhaps use more meaningful names than just a letter s

mamayuan commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,28 @@
+    public static String getExitMessage() {
+        return "Bye. Hope to see you again soon bro!";
+    }
+
```
whitespace here as well

mamayuan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,80 @@
+        try {
+            // create data directory if it does not exist
+            Files.createDirectories(Paths.get("./data"));
+            File f = new File(filePath);
```
perhaps use file as the variable name rather than just f?

samuelmui8 committed:
Add assertions

There are no checks to ensure certain assumptions in the code holds.

Assertions have been added to document important assumptions that should
hold at various points in the code.

Assertions are used to define assumptions about the program state so
that the runtime can verify them.

samuelmui8 committed:
Make the happy path prominent

Several methods have nested happy paths.

Use guard clauses to reduce nesting of happy paths.

Happy path is clearer and code is more readable.

samuelmui8 committed:
Shorten a long method

The stringToTask method exceeds 30 LoC.

Its functionality is abstracted to 3 separate methods.

The method is shorter and is more easily processed by the reader.

samuelmui8 committed:
Merge pull request #2 from samuelmui8/branch-A-Assertions

Add assertions

samuelmui8 committed:
Merge branch 'master' of https://github.com/samuelmui8/ip

samuelmui8 committed:
Merge pull request #1 from samuelmui8/branch-A-CodeQuality

Branch a code quality

samuelmui8 committed:
Merge branch 'master' of https://github.com/samuelmui8/ip

samuelmui8 committed:
Add ability to sort deadlines and events

When there are many different types of tasks in the main list, it is
difficult to see which deadlines and events are coming up.

Allow users to sort deadlines and events by date so that they can see
which ones are approaching.

This helps to improve the usability of the bot.

samuelmui8 committed:
Merge branch 'branch-C-Sort'

samuelmui8 committed:
Update assertions and javadocs

samuelmui8 committed:
Gave the chatbot a dark theme

samuelmui8 committed:
Merge branch 'branch-A-BetterGui'

samuelmui8 committed:
Add window header and UI.png

samuelmui8 committed:
Fixed naming of Ui.png

samuelmui8 committed:
Abstract UI interactions into UI class

samuelmui8 committed:
Bug fixes to for index out of bounds error

samuelmui8 committed:
Update README.md

samuelmui8 committed:
Fix bug where label cuts off

samuelmui8 committed:
Update README.md
